src/plugins/keras_support/switch_operation.cpp

```diff
@@ -0,0 +1,397 @@
+                thenf (statement) : a statement
+                elsef (statement, optional) : a statement
+
+            Returns:
+
+              The value returned by the statement that was executed, `nil`
+              otherwise)"
+            )
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    switch_operation::switch_operation(
+            primitive_arguments_type&& operands,
+            std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    ///////////////////////////////////////////////////////////////////////////
+    bool switch_operation::validate_shapes(std::size_t const& ndims_cond,
+        std::size_t const& ndims_then,
+        std::array<std::size_t, PHYLANX_MAX_DIMENSIONS>&& dims_cond,
+        std::array<std::size_t, PHYLANX_MAX_DIMENSIONS>&& dims_then) const
+    {
+        if (ndims_cond <= ndims_then)
+        {
+            if (dims_cond == dims_then)
+                return true;
+
+            if (ndims_cond == ndims_then)
+            {
+                for (std::size_t i = 0; i != ndims_cond; ++i)
+                    if (dims_cond[i] != dims_then[i])
+                        if (dims_cond[i] != 1)
+                            return false;
+                return true;
+            }
+
+            if (ndims_cond < ndims_then)
+            {
+                if (ndims_cond == 0)
+                    return true;
+
+                switch (ndims_then)
+                {
+                case 1:
+                {
+                    if (dims_cond[0] == 1)
+                        return true;
+                    break;
+                }
+
+                case 2:
+                {
+                    if (ndims_cond == 1)
+                        if (dims_cond[0] == dims_then[0] || dims_cond[0] == 1)
+                            return true;
+                    break;
+                }
+
+                case 3:
+                    if (ndims_cond == 1)
+                    {
+                        if (dims_cond[0] == dims_then[0] || dims_cond[0] == 1)
+                            return true;
+                    }
+                    else if (ndims_cond == 2)
+                    {
+                        if ((dims_cond[0] == dims_then[0] ||
+                                dims_cond[0] == 1) &&
+                            (dims_cond[1] == dims_then[1] || dims_cond[1] == 1))
+                            return true;
+                    }
+                    break;
+
+                default:
+                    break;
+                }
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "switch_operation::validate_shapes",
+                    util::generate_error_message(
+                        "the condition shape is not compatible with the "
+                        "`then_expression` shape",
+                        name_, codename_));
```

Comment:
This exception can be removed as the code falls through directly into an equivalent one.